scvi/dataloaders/_data_splitting.py

```diff
@@ -88,14 +89,21 @@ def __init__(
         adata_manager: AnnDataManager,
         train_size: float = 0.9,
         validation_size: Optional[float] = None,
-        use_gpu: bool = False,
+        use_gpu: Optional[Union[str, int, bool]] = None,
         **kwargs,
     ):
         super().__init__()
         self.adata_manager = adata_manager
         self.train_size = float(train_size)
         self.validation_size = validation_size
         self.data_loader_kwargs = kwargs
+
+        if use_gpu is not None:
+            warnings.warn(
+                "The `use_gpu` argument is deprecated in v0.20 and will be removed in v1.0. "
+                "Please directly pass in `pin_memory` directly for device memory pinning.",
+                DeprecationWarning,
+            )
```

Comment:
this will cause a warning for anyone using a gpu right? I think most users don't go this deep into the code, nor do I think anyone is actively turning on this pin memory feature.

I think it's safe (although breaking) to remove this tiny feature altogether and expose pin memory explicitly here with default equal to False. We can then

1. deprecate `settings.dl_pin_memory_gpu_training` if someone tries to set it
2. and expose data_splitter_kwargs in the train method